driver-core/src/main/java/com/datastax/driver/core/TypeCodec.java

```diff
@@ -508,6 +508,14 @@ public DataType getCqlType() {
     /**
      * Return {@code true} if this codec is capable of serializing
      * the given {@code javaType}.
+     * <p>
+     * The default implementation is <em>invariant</em> with respect to the passed
+     * argument (through the usage of {@link TypeToken#equals(Object)}.
+     * This means that, by default, a codec will only ever accept the
+     * <em>exact</em> Java type that it has been created for.
```

Comment:
"by default" implies that you can override it. I can't think of any legitimate case where you would do something different in a subclass (without wreaking havoc), so maybe this should be more like "this is invariant and it's strongly recommended not to override it". Or maybe the method should even be final...